File: programmers/lv2/greedy/구명보트.py
from collections import defaultdict

def solution(people, limit):
    answer, visited = 0, [False for _ in range(len(people))]
    people.sort()
    
    l, r = 0, len(people) - 1
    while l <= r:
        if l == r: 
            answer += 1
            break
        if people[l] + people[r] <= limit:
            l += 1
        r -= 1
        answer += 1
        
    return answer

File: programmers/lv2/greedy/구명보트.js
function solution(people, limit) {
    var answer = 0;
    
    people.sort((a, b) => a - b);
    var l = 0, r = people.length - 1;
    
    while(l <= r) {
        if(people[l] + people[r] <= limit) {
            l++;
        }
        r--;
        
        answer++;
    }
    
    return answer;
}
